.github/workflows/openapi-diff.yml

```diff
@@ -1,42 +1,62 @@
 name: openapi-diff
 on:
   pull_request:
-    paths:
-      - 'api-docs/**'
 
 jobs:
-    check-diff:
-      runs-on: ubuntu-18.04
-      steps:
-        - name: Checkout getsentry/sentry
-          uses: actions/checkout@v2
-          with:
-            path: sentry
-
-        - name: Getsentry Token
-          id: getsentry
-          uses: getsentry/action-github-app-token@v1
-          with:
-            app_id: ${{ secrets.SENTRY_INTERNAL_APP_ID }}
-            private_key: ${{ secrets.SENTRY_INTERNAL_APP_PRIVATE_KEY }}
-
-        - name: Checkout getsentry/sentry-api-schema
-          uses: actions/checkout@v2
-          with:
-            ref: 'main'
-            repository: getsentry/sentry-api-schema
-            path: sentry-api-schema
-            token: ${{ steps.getsentry.outputs.token }}
-
-        - name: Install/setup node
-          uses: volta-cli/action@v1
-
-        - name: Build OpenAPI Derefed JSON
-          run: |
-            cd sentry
-            yarn install --frozen-lockfile
-            yarn run build-derefed-docs api-docs/openapi-derefed.json
-
-        - name: Compare OpenAPI Derefed JSON
-          run: |
-            npx json-diff --color sentry-api-schema/openapi-derefed.json sentry/api-docs/openapi-derefed.json
+  check-diff:
+    name: build api
+    runs-on: ubuntu-20.04
+    timeout-minutes: 90
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+
+    steps:
+      - name: Getsentry Token
+        id: getsentry
+        uses: getsentry/action-github-app-token@v1
+        with:
+          app_id: ${{ secrets.SENTRY_INTERNAL_APP_ID }}
+          private_key: ${{ secrets.SENTRY_INTERNAL_APP_PRIVATE_KEY }}
+
+      - uses: actions/checkout@v2
+        with:
+          # Avoid codecov error message related to SHA resolution:
+          # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
+          fetch-depth: '2'
+
+      - name: Check for python file changes
+        uses: getsentry/paths-filter@v2
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
+
+      - name: Setup sentry env (python ${{ matrix.python-version }})
```

Comment:
we need the python environment to be set up now since spectacular needs it to build the JSON, i assumed this step copy pasted from backend should suffice.